Mathlib/MeasureTheory/Decomposition/IntegralRNDeriv.lean

```diff
@@ -0,0 +1,94 @@
+/-
+Copyright (c) 2025 Rémy Degenne. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Rémy Degenne, Lorenzo Luccioli
+-/
+import Mathlib.Analysis.Convex.Integral
+import Mathlib.MeasureTheory.Decomposition.RadonNikodym
+
+/-!
+# Integrals of functions of Radon-Nikodym derivatives
+
+## Main statements
+
+* `mul_le_integral_rnDeriv_of_ac`: for a convex continuous function `f` on `[0, ∞)`, if `μ`
+  is absolutely continuous with respect to `ν`, then
+  `(ν univ).toReal * f ((μ univ).toReal / (ν univ).toReal) ≤ ∫ x, f (μ.rnDeriv ν x).toReal ∂ν`.
+
+-/
+
+
+open Set
+
+namespace MeasureTheory
+
+variable {α : Type*} {mα : MeasurableSpace α} {μ ν : Measure α} {f : ℝ → ℝ}
+
+/-- For a convex continuous function `f` on `[0, ∞)`, if `μ` is absolutely continuous
+with respect to a probability measure `ν`, then
+`f (μ univ).toReal ≤ ∫ x, f (μ.rnDeriv ν x).toReal ∂ν`.
+Note that the `ContinuousOn f (Ici 0)` hypothesis is really only about continuity at 0,
+since the convexity gives continuity elsewhere. -/
```

Comment:
Should the condition be `ContinuousWithinAt f (Ici 0) 0` then?

Reply:
In the project where this is coming from, we used continuity on the whole interval everywhere, in order to avoid having to switch from one type of continuity statement to another, but I agree it would be better to require the minimal hypothesis. I changed it.